torchgeo/datasets/cdl.py

```diff
@@ -292,8 +294,7 @@ def _verify(self) -> None:
         exists = []
         for year in self.years:
             filename_year = self.filename_glob.replace("*", str(year))
-            pathname = os.path.join(self.root, "**", filename_year)
-            for fname in glob.iglob(pathname, recursive=True):
+            for fname in self.list_files(filename_glob=filename_year):
```

Comment:
@adamjstewart This is also a use case where overriding the `filename_glob` is proposed.

Reply:
In this case `self.list_files` isn't significantly better than `glob.iglob`, you're only saving 1 line of code and the result is that it can't be a property anymore.